x/blob/client/cli/wirepayfordata.go

```diff
@@ -54,11 +58,89 @@ func CmdWirePayForBlob() *cobra.Command {
 				return err
 			}
 
-			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), pfbMsg)
+			txBytes, err := writeTx(clientCtx, sdktx.NewFactoryCLI(clientCtx, cmd.Flags()), pfbMsg)
+			if err != nil {
+				return err
+			}
+
+			blobTx, err := coretypes.MarshalBlobTx(txBytes, blob)
+			if err != nil {
+				return err
+			}
+
+			// broadcast to a Tendermint node
+			res, err := clientCtx.BroadcastTx(blobTx)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
 		},
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
 
 	return cmd
 }
+
+// writeTx attempts to generate and sign a transaction using the normal
+// cosmos-sdk cli argument parsing code with the given set of messages. It will also simulate gas
+// requirements if necessary. It will return an error upon failure.
+//
+// NOTE: Copy paste forked from the cosmos-sdk so that we can wrap the PFB with
+// a blob while still using all of the normal cli parsing code
+func writeTx(clientCtx client.Context, txf sdktx.Factory, msgs ...sdk.Msg) ([]byte, error) {
```

Comment:
had to copy and paste this from the cosmos-sdk so that we can wrap the tx with a blob before broadcasting it.